Sources/HStack.swift

```diff
@@ -59,24 +55,47 @@ open class HStack: Stack {
                 frames.append(frame)
                 x += stackableWidth
             }
+            
+            x += self.spacing
         }
         
         return frames
     }
     
+    public var intrinsicContentSize: CGSize {
+        // TODO: add adjustments for layoutMargins (not currently needed so okay to defer)
+        
+        let items = visibleThingsToStack()
+        
+        guard !items.isEmpty else { return .zero }
+        
+        // width
+        let totalWidthOfItems = items.reduce(0, { $0 + $1.intrinsicContentSize.width })
+        let totalHorizontalSpacing = max(CGFloat(items.count) - 1, 0) * spacing
+        let intrinsicWidth = totalWidthOfItems + totalHorizontalSpacing
```

Comment:
Should we moves the width and height calc out to private funcs?

Reply:
I can, I didn't think the logic was too complicated in this computed property

Reply:
I actually might leave it how it is, if thats cool, I moved it out to private funcs and it seemed a bit messier to me

Reply:
Ok no worries.